packages/workspace-tools/src/getPackageInfos.ts

```diff
@@ -1,69 +1,53 @@
+import { getWorkspaces, getWorkspacesAsync } from "./workspaces/getWorkspaces";
 
 export function getPackageInfos(cwd: string) {
-  let packageJsonFiles = getAllPackageJsonFiles(cwd);
-
-  if (packageJsonFiles.length === 0 && fs.existsSync(path.join(cwd, "package.json"))) {
-    packageJsonFiles = [path.join(cwd, "package.json")];
-  }
-
   const packageInfos: PackageInfos = {};
-  if (packageJsonFiles && packageJsonFiles.length > 0) {
-    packageJsonFiles.forEach((packageJsonPath: string) => {
-      try {
-        const packageJson = JSON.parse(fs.readFileSync(packageJsonPath, "utf-8"));
-        packageInfos[packageJson.name] = infoFromPackageJson(packageJson, packageJsonPath);
-      } catch (e) {
-        if (e instanceof Error) {
-          // Pass, the package.json is invalid
-          throw new Error(`Invalid package.json file detected ${packageJsonPath}: ${e.message}`);
-        } else {
-          throw e;
-        }
-      }
-    });
-    return packageInfos;
+  const workspacePackages = getWorkspaces(cwd);
+
+  if (workspacePackages.length) {
+    for (const pkg of workspacePackages) {
+      packageInfos[pkg.name] = pkg.packageJson;
+    }
+  } else {
+    const rootInfo = tryReadRootPackageJson(cwd);
+    if (rootInfo) {
+      packageInfos[rootInfo.name] = rootInfo;
+    }
   }
 
-  return {};
+  return packageInfos;
 }
 
 export async function getPackageInfosAsync(cwd: string) {
-  let packageJsonFiles = await getAllPackageJsonFilesAsync(cwd);
+  const packageInfos: PackageInfos = {};
+  const workspacePackages = await getWorkspacesAsync(cwd);
 
-  if (packageJsonFiles.length === 0 && fs.existsSync(path.join(cwd, "package.json"))) {
-    packageJsonFiles = [path.join(cwd, "package.json")];
+  if (workspacePackages.length) {
+    for (const pkg of workspacePackages) {
+      packageInfos[pkg.name] = pkg.packageJson;
+    }
+  } else {
+    const rootInfo = tryReadRootPackageJson(cwd);
+    if (rootInfo) {
+      packageInfos[rootInfo.name] = rootInfo;
+    }
   }
 
-  const packageInfos: PackageInfos = {};
-
-  if (packageJsonFiles && packageJsonFiles.length > 0) {
-    const packageInfoPromises = packageJsonFiles.map(async (packageJsonPath: string) => {
-      try {
-        const packageJson = JSON.parse(await fsPromises.readFile(packageJsonPath, "utf-8"));
-        return infoFromPackageJson(packageJson, packageJsonPath);
-      } catch (e) {
-        if (e instanceof Error) {
-          // Pass, the package.json is invalid
-          throw new Error(`Invalid package.json file detected ${packageJsonPath}: ${e.message}`);
-        } else {
-          throw e;
-        }
-      }
-    });
+  return packageInfos;
+}
 
-    const results = await Promise.all(packageInfoPromises);
-    for (const packageInfo of results) {
-      packageInfos[packageInfo.name] = packageInfo;
+function tryReadRootPackageJson(cwd: string): PackageInfo | undefined {
```

Comment:
can we have an async verison of this? It gave me a bit of a boost when we read these files in parallel as determined by libuv